internal/ignition/ignition.go

```diff
@@ -205,42 +146,30 @@ func (g *installerGenerator) generateOcsManifests(ocsValidatorConfig *ocs.Config
 	return nil
 }
 
-func (g *installerGenerator) generateOperatorsManifests(ctx context.Context, installerPath string, envVars []string, ocsValidatorConfig *ocs.Config) error {
-	lsoEnabled := false
-	ocsEnabled := g.checkOcsEnabled()
-	if ocsEnabled {
-		lsoEnabled = true // if OCS is enabled, LSO must be enabled by default
-	} else {
-		lsoEnabled = g.checkLsoEnabled()
+func (g *installerGenerator) writeOperatorManifests(installerPath string, envVars []string) error {
+	if !g.operatorsManager.AnyOperatorEnabled(g.cluster) {
+		return nil
 	}
-	if lsoEnabled || ocsEnabled {
-		err := g.createManifestDirectory(installerPath, envVars)
-		if err != nil {
-			g.log.Error("Failed to create Manifest directory ", err)
-			return err
-		}
+	err := g.createManifestDirectory(installerPath, envVars)
```

Comment:
I have few comments about this which aren't related to the PR, but for the way manifests are generated for the operators:

This command (which actually runs `openshift-install --dir $INSTALL_DIR create manifests`) is part of the ignition generation flow - https://github.com/openshift/assisted-service/blob/master/internal/ignition/ignition.go#L298

Invoking it here seems to be out of context, since AFAIU it was supposed to be invoked if there was indication of manifests for the cluster, but not for storing generated manifests directly.

What I had in mind is the following sequence:
0. Add manifests to cluster using the manifest API (e.g. https://github.com/openshift/assisted-service/blob/master/internal/network/manifests_generator.go#L129)
1.  Generate `install-config.yaml`
2. Write `install-config.yaml` to file-system (this is equivalent to `openshift-install --dir $INSTALL_DIR create install-config`)
3. Consider manifests if exists (`openshift-install --dir $INSTALL_DIR create manifests`)
4. Generate ignitions (`openshift-install --dir $INSTALL_DIR create ignition-configs`)
5. Use generated ignition files for bootstrapping the nodes

While the current implemented sequence invokes step 0 after step 3.
In addition, operator manifests are being written to file-system directly and are not going through the manifests api.
Specifically function generateLsoManifests from https://github.com/openshift/assisted-service/blob/ca8cc1b60e8b885f8341908f6afa21fae6cdeda1/internal/ignition/ignition.go#L196
re-implements https://github.com/openshift/assisted-service/blob/2cf921c2df5762ab952f7cdb4f3988480e7b11e7/internal/manifests/manifests.go#L47

Going through the manifest API will align the code to use a single method of creating the manifests and will make sure that any generated manifests will remain available to the user in any file system implementation (FS, S3) and will allow to debug potential issues with the installed operators.

I think the current implementation should be changed to use the manifests via its designated API, similar to other internal usages of it.

@gamli75

Reply:
I assumed that MGMT-3371 won't change any logic currently present in the code and I think that changes you propose should be introduced only after the MGMT-3371 is complete (all planned PRs are merged). WDYT?

Reply:
@masayag can you open a different ticket on this?

Reply:
Created [MGMT-4188](https://issues.redhat.com/browse/MGMT-4188)